src/jdk.jcmd/share/man/jcmd.1

```diff
@@ -254,13 +253,9 @@ Impact: Low
 .PP
 \f[I]arguments\f[R]:
 .IP \[bu] 2
-\f[I]filename\f[R]: (Optional) The name of the map file (STRING,
-/tmp/perf-<pid>.map)
-.PP
-If \f[V]filename\f[R] is not specified, a default file name is chosen
-using the pid of the target JVM process.
-For example, if the pid is \f[V]12345\f[R], then the default
-\f[V]filename\f[R] will be \f[V]/tmp/perf-12345.map\f[R].
+\f[I]filename\f[R]: (Optional) The name of the map file.
+If %p is specified in the filename, it is expanded to the JVM\[aq]s PID.
+(FILE, \[dq]/tmp/perf-%p.map\[dq])
```

Comment:
I'm not sure about this change to FILE from STRING. Are there previous uses of FILE? We still seem to use STRING for JFR.dump, JFR.start, and JFR.stop.

Reply:
OK yes - my thinking is:

They are strings of course, but they are also names of an output FILE, and those are distinct because they do %p substitution.

They need to have a "type" of FILE internally in the tool, so they get %p substitution.

The FILE instead of STRING doesn't need to be in the man page, but it would make sense that the man page is the same as the live help output as much as possible.

After the intro, the body of the man page mostly duplicates the online help for the commands.
So much so that you'd think we can automate that, which would make a great future step.
Then we would get FILE for the type of these arguments, not STRING.  So I used FILE here as it avoids a future issue.

I think we should move other output file names e.g. in the JFR commands to FILE over time, after verifying they do the %p substitution.  There could be %t timestamp as well, I think an enhancement request for that exists.

Then the docs could summarise everything about the FILE parameters and not have to duplicate it every time.

Reply:
Ok. I see now that [JDK-8334492](https://bugs.openjdk.org/browse/JDK-8334492) made the STRING -> FILE change, so yes the docs should be consistent, but I think we need a new CR filed to cleanup JFR dcmds, which continue to use STRING. This probably should have been done with [JDK-8334492](https://bugs.openjdk.org/browse/JDK-8334492).

Reply:
Thanks, yes should have made this clearer and yes will follow up on the additional commands, we should transition them to the same language.

Reply:
JFR.dump, JFR.start, and JFR.stop are written in Java and have separated from the Hotspot dcmd infrastructure. For example, this is how file expansion is handled by JFR.start (and -XX:StartFlightRecording):

    String path = expandFilename(parser.getOption("filename"));

Default values are hardwired into the command and the help is custom. The command allows additional options, like event settings. that are not "DCmdArguments". One could replace "STRING" with "FILE", but the only visible impact would be the metadata provided by the DiagnosticCommandMBean.

Reply:
I filed an issue:
https://bugs.openjdk.org/browse/JDK-8338405